src/test/java/io/vertx/core/http/Http2ServerTest.java

```diff
@@ -1278,6 +1279,26 @@ public void onPushPromiseRead(ChannelHandlerContext ctx, int streamId, int promi
     await();
   }
 
+  @Test
+  public void testHostHeaderInsteadOfAuthorityPseudoHeader() throws Exception {
+    testValidRequestHeaders(new DefaultHttp2Headers().method("GET").scheme("https").path("/").set("host", DEFAULT_HTTPS_HOST_AND_PORT));
+  }
+
+  private void testValidRequestHeaders(Http2Headers headers) throws Exception {
+    server.requestHandler(req -> {
+      testComplete();
```

Comment:
I think we need to check the authority value here is present and its value

Reply:
good point. done.